server.py

```diff
@@ -12,28 +12,37 @@
     "player": None
 }
 
-@app.route("/")
-def index():
-    return "Welcome to tennis reservations"
 
-@app.route('/reservations', methods = ['GET', 'POST'])
-def api_reservations():
-    if request.method == 'GET':
-        if request.args.get('hour'):
-            playtime = request.args.get('hour')
-            js = json.dumps(RESERVATIONS[playtime])
-            resp = Response(js, status=200, mimetype='application/json')
-            return resp
-        else:
-            js = json.dumps(RESERVATIONS)
-            resp = Response(js, status=200, mimetype='application/json')
-            return resp
 
-    elif request.method == 'POST':
+@app.route('/reservations', methods = ['GET'])
+def reservations_get():
+    if request.args.get('hour') not in RESERVATIONS.keys() and request.args.get('hour') is not None:
+        resp = Response(response='Invalid hour format', status=400)
+        return resp
+    elif request.args.get('hour') in RESERVATIONS.keys():
+        playtime = request.args.get('hour')
+        js = json.dumps(RESERVATIONS[playtime])
+        resp = Response(js, status=200, mimetype='application/json')
+        return resp
+    elif request.args.get('hour') is None:
+        js = json.dumps(RESERVATIONS)
+        resp = Response(js, status=200, mimetype='application/json')
+        return resp
+
+@app.route('/reservations', methods = ['POST'])
+def reservations_post():
+    if request.get_json() is None:
+        resp = Response(status = 400, response='Invalid body')
+        return resp
+    elif request.args.get('hour') not in RESERVATIONS.keys():
```

Comment:
POST requests don't have query parameters so `request.args.get('hour')` won't work. You need to get the hour out from the JSON payload.